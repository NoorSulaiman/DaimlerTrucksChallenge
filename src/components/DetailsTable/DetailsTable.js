import React from "react";
import {
  Table,
  TableContainer,
  TableBody,
  TableRow,
  TableCell,
  Paper,
} from "@mui/material";

export default function DetailsTable({ component }) {
  const componentProps = Object.keys(component).map((prop, indx) => {
    return (
      <TableRow key={component}>
        <TableCell component="th" scope="row">
          {prop}
        </TableCell>
        <TableCell align="right">{JSON.stringify(component[prop])}</TableCell>
      </TableRow>
    );
  });
  return (
    <div>
      <TableContainer component={Paper}>
        <Table aria-label="details table">
          <TableBody>{componentProps}</TableBody>
        </Table>
      </TableContainer>
    </div>
  );
}
