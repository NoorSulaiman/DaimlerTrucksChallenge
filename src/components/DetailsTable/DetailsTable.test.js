import React from "react";
import { shallow } from "enzyme";
import DetailsTable from "./DetailsTable";

const componentObj = {
  panelNumber: "22-58883-000",
  panelDescription: "PNL-INR TRIMPLTKAHLUA SPRKL",
  location: "AGAAA",
  parentPanelNumber: "734-C02202",
  modelCode: "H",
  type: "hardwired-component",
  switchId: "",
  bom: "734-C02202",
};

describe("DetailsTable", () => {
  it("should render DetailsTable correctly with given props", () => {
    const component = shallow(<DetailsTable component={componentObj} />);
    expect(component).toMatchSnapshot();
  });
});
