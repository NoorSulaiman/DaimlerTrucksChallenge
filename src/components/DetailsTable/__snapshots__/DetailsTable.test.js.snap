// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsTable should render DetailsTable correctly with given props 1`] = `ShallowWrapper {}`;
